### ISSUE TITLE
Is possible to do this nowdays with Gnome/X11/libinput ?

### ISSUE THREAD

**Issue by grigio** [quadpixels/three-finger-dragging#2]
as above

**Comment by quadpixels** [comment]
Hello, Grigio,
I haven't looked into this code for some time but people maintaining freedesktop said:

> See comment #10 for the summary, we're pushing this to the compositor.

(See here: https://bugs.freedesktop.org/show_bug.cgi?id=89999)

**Comment by grigio** [comment]
Thanks for the update, I'll follow that

**Comment by rickyzhang82** [comment]
No, there won't be anything. Check my recent comment: https://bugzilla.gnome.org/show_bug.cgi?id=768421#c5

We need to figure it out the way by OURSELVES. Guys, are you interested in getting this done?

**Comment by quadpixels** [comment]
Ah..... I stopped working on this thing last year because I need to get graduate school stuff done && some other obligations. ლ(ಠ益ಠლ)
Also as I remember the code in this repository still have some minor bugs that have to be fixed.
Maybe I am a bit too lazy and just lack a push to bring this thing further (to production quality.)

Thanks Ricky for the comments, I'll check out TigerSoldier's hacks in LibInput here. [https://github.com/tigersoldier/libinput/tree/wip/three-fingers-dragging](url)

**Comment by rickyzhang82** [comment]
The link you post is dead.

Someone suggest to use bounty to motivate them to fix: https://bugzilla.gnome.org/show_bug.cgi?id=768421#c10

That's not a bad idea. It will take me at least a month of spare time to figure it out and patch it.

**Comment by quadpixels** [comment]
@rickyzhang82 Sorry for the bad link, I fixed the link. It should point to [https://github.com/tigersoldier/libinput/tree/wip/three-fingers-dragging](https://github.com/tigersoldier/libinput/tree/wip/three-fingers-dragging) .

**Comment by schlomo** [comment]
See also this libinput bug https://bugs.freedesktop.org/show_bug.cgi?id=89999 where I just now argued in favor of this.

**Comment by schlomo** [comment]
The sad thing is that on Ubuntu 16.04 there are actually three drivers for touchpad:

- `xserver-xorg-input-synaptics-hwe-16.04`
- `xserver-xorg-input-mtrack-hwe-16.04`
- `xserver-xorg-input-libinput-hwe-16.04`

I tried all of them on a [Lenovo T460s](https://certification.ubuntu.com/hardware/201512-20435/) but none of them supports three-finger-drag out of the box. However, the touchpad does support three and four finger guestures (use `libinput-debug-events` to observe swipe and pinch gestures).

**Comment by quadpixels** [comment]
@schlomo Thank you for your argument and I've also seen the replies in bug 89999 regarding libinput and composer ---- I don't quite like the idea of filing a bug in every compositor, but it did look like Ubuntu Unity has some 4-finger dragging enabled in as early as 2014~2015. So this is worth looking into.
I would probably argue that having 3 finger drag in the driver provides some "default behavior" for the 3-finger related gestures/drags, but that would depend on how slowly or how fast the compositors adopt 3 finger drags.

Anyway, I have fixed some bugs in the current code; I haven't looked at it for about a year and this time I found a few problems which would make users think this thing is totally useless. Hopefully after the fixes in the past few days they would be less likely to think so. :-S
